### PR TITLE
[r] Data refresh for new shapes

### PR DESCRIPTION
**Pull request by johnkerl**
**Issue and/or context:** As tracked on issue #2407 / [[sc-51048]](https://app.shortcut.com/tiledb-inc/story/51048).

Note that the intended Python and R API changes are all agreed on and finalized as described in #2407.

**Changes:**

* Regenerate data in the `inst/extdata` directory to have new-style shapes -- primarily for the benefit of vignettes on #3302.
* Add a missing script for regenerating one of the two datasets
* Add `README.md` for the benefit of future maintainers

**Notes for Reviewer:**

### CODE REVIEW

File: apis/r/tests/testthat/test-example-datasets.R [hunk 1]
```diff
@@ -1,7 +1,7 @@
 test_that("example dataset access", {
   expect_length(
     list_datasets(),
-    length(dir(example_data_dir()))
+    length(dir(example_data_dir())) - 1 # There is a README.md
```
Comment (mojaveazure) [inline apis/r/tests/testthat/test-example-datasets.R:4]: I don't see a place in `README.md` that explains why this `- 1` exists

**Review by mojaveazure — CHANGES_REQUESTED** [review]
So it seems the original `pbmc3k-processed-obs` was from Scanpy, not Seurat (based on the column name changes in `test-SOMAArrayReader-Arrow.R` and `test-SOMADataFrame.R`; I've got one comment about a missing `README` entry

File: apis/r/tests/testthat/test-example-datasets.R (hunk 1 above)
Comment (johnkerl) [reply apis/r/tests/testthat/test-example-datasets.R:4]: The `README.md` file is an item within `example_data_dir`.

Comment (johnkerl) [reply apis/r/tests/testthat/test-example-datasets.R:4]: I can say that in the `README.md` file.

Comment (johnkerl) [reply apis/r/tests/testthat/test-example-datasets.R:4]: @mojaveazure https://github.com/single-cell-data/TileDB-SOMA/pull/3303/commits/b6c59b4c72142a6fd01f4110e08cee0ccf9a45b8

**Review by mojaveazure — APPROVED** [review]

**Pull request merged**